solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -119,10 +128,18 @@
   private final int queryResultMaxDocsCached;
   private final boolean useFilterForSortedQuery;
 
+  /**
+   * Special-case cache to handle the lazy-init of {@link #liveDocs}.
+   */
+  private final SolrCache<MatchAllDocsQuery,BitDocSet> liveDocsCache;
```

Comment:
Yeah; it started out as something ostensibly "simpler" like that, but once you layer in the metrics (mainly to support tests) it becomes less simple. I wonder how succinct it could be, maybe worth a shot.

I'll mull this; if it's something you feel (somewhat) strongly about I'll take a crack at it.

Reply:
I don't feel strongly about it, and would be satisfied with an additional line in the comment saying why it is what it is.